client/typescript/fine-uploader.d.ts

```diff
@@ -2984,397 +2315,1240 @@ declare namespace FineUploader {
+             *
+             * ###Note:
+             * Fine Uploader will still send the `bucket`, `name`, `key`, `filename`, `UUID`, and `etag` (if available) as well
+             *
+             * @param object newParams : The additional parameters set for the upload request
+             * @param number id : A file id to apply these upload success parameters to
+             */
+            setUploadSuccessParams(newParams: any, id?: number): void;
+
+        }
+
+
+        /**
+         * S3CredentialsOptions
+         */
+        export interface S3CredentialsOptions {
+            /**
+             * Temporary public AWS key
+             *
+             * @default `null`
+             */
+            accessKey: string;
+            /**
+             * Expiration date for temporary credentials. May be an ISO 8601 String or a `Date` object.
+             *
+             * @default `null`
+             */
+            expiration: string | Date;
+            /**
+             * Temporary secret AWS key
+             *
+             * @default `null`
+             */
+            secretKey: string;
+            /**
+             * Session token associated with the temporary credentials
+             *
+             * @default `null`
+             */
+            sessionToken: string;
+        }
+
+        /**
+         * S3ChunkingOptions
+         */
+        export interface S3ChunkingOptions extends ChunkingOptions {
+            /**
+             * The maximum size of each part, in bytes
+             *
+             * @default `5242880`
+             */
+            partSize: number;
+        }
+
+        /**
+         * S3CorsOptions
+         */
+        export interface S3CorsOptions extends CorsOptions {
+            /**
+             * Enables or disables cross-domain ajax calls (if the `expected` property is true) in IE9 and older.
+             *
+             * @default `true`
+             */
+            allowXdr: boolean;
+        }
+
+        /**
+         * S3iFrameSupportOptions
+         */
+        export interface S3iFrameSupportOptions {
+            /**
+             * This is required if you plan on supporting browsers that do not implement the File API, such as IE9 and older.
+             * This must point to a blank page on the same origin/domain as the page hosting Fine Uploader
+             *
+             * @default `null`
+             */
+            localBlankPagePath: string;
+        }
+
+        /**
+         * type for S3's bucket object property
+         */
+        export interface BucketFunction {
+            constructor(id: number);
+        }
+
+        /**
+         * type for S3's host object property
+         */
+        export interface HostFunction {
+            constructor(id: number);
+        }
+
+        /**
+         * type for S3's key object property
+         */
+        export interface KeyFunction {
+            constructor(id: number);
```

Comment:
@jclangst this function and 3 above it has declaration like this `(id: number): PromiseOptions | string;` before.
Any reason for the change?